.github/workflows/build.yml

```diff
@@ -15,9 +15,9 @@ jobs:
         java: [8, 11, 17, 21]
       fail-fast: true
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
       - name: Install Java ${{ matrix.java }}
-        uses: actions/setup-java@v2
+        uses: actions/setup-java@v3
         with:
           java-version: ${{ matrix.java }}
           distribution: 'adopt'
```

Comment:
"adopt" can be updated to "temurin".

Reply:
Already changed in this pull request as far as I see. Waiting on Temurin 21 release is necessary...

Reply:
which is waiting because of some stupid legal bullshit, maybe it should be changed to a vendor that actually releases timely, since this isnt the first case where the eclipse foundation is slow.

Reply:
> which is waiting because of some stupid legal bullshit, maybe it should be changed to a vendor that actually releases timely, since this isnt the first case where the eclipse foundation is slow.

Yes, the Eclipse foundation is very slow. I think vendors like Azul and the others had to do the same legal things and got it done on time. Maybe switch to Azul Zulu?

Reply:
done